workspaces/liferay-ticket-workspace/client-extensions/liferay-ticket-custom-element/src/components/TicketPreview.tsx

```diff
@@ -0,0 +1,179 @@
+	'M2H8_SUPPORT_SAM': icons.SAM,
+};
+
+const getUserIcon = (ticket: Ticket): string => {
+	if (ticket.assignee) {
+		let icon = usersIconsMap[ticket.assignee.externalReferenceCode];
+		if (!icon) {
+			icon = usersIconsMap[''];
+		}
+		else {
+			return icon;
+		}
+	}
+
+	return usersIconsMap[''];
+};
+
+const TicketPreview: React.FC<{queryClient: QueryClient; ticket: Ticket}> = ({
+	queryClient,
+	ticket,
+}) => {
+	const [isLoading, setIsLoading] = useState(false);
+
+	const {
+		attributes,
+		isDragging,
+		listeners,
+		setNodeRef,
+		transform,
+	} = useDraggable({data: ticket, id: ticket.id + '_draggable'});
+
+	const style: React.CSSProperties = {
+		backgroundColor: isDragging ? '#F4F9F9' : 'white',
+		position: 'relative',
+		transform: CSS.Translate.toString(transform),
+		zIndex: isDragging ? 2147483647 : 1,
+	};
+
+	const mutation = useMutation({
+		mutationFn: (ticket: Ticket) => {
+			setIsLoading(true);
+
+			return assignToMe(ticket);
+		},
+		onError: () => {
+			setIsLoading(false);
+			Liferay.Util.openToast({
+				message: 'Something went wrong, please try again.',
+				type: 'danger',
+			});
+		},
+		onSuccess: (responseCode) => {
+			setIsLoading(false);
+			queryClient.invalidateQueries();
+			if (responseCode === 200) {
+				Liferay.Util.openToast({
+					message: 'Ticket was assigned to you successfully!',
+					type: 'success',
+				});
+			}
+			else {
+				Liferay.Util.openToast({
+					message: 'Something went wrong, please try again.',
+					type: 'danger',
+				});
+			}
+		},
+	});
+
+	return (
+		<div
+			ref={setNodeRef}
+			{...listeners}
+			{...attributes}
+			className="bg-white border-light mb-3 rounded ticket-preview-container"
+			style={style}
+		>
+			<ClayPanel
+				collapsable
+				displayTitle={
+					<ClayPanel.Title>
+						<h5 className="text-dark ticket-subject">
+							{ticket.subject}
+						</h5>
+						<div className="mt-3">
+							<img
+								className="rounded-circle"
+								height={25}
+								src={getUserIcon(ticket)}
+								width={25}
+							/>
+							<i className="ml-2">
+								{ticket.assignee
+									? ticket.assignee?.familyName
+									: 'Not assigned'}
+							</i>
+						</div>
+					</ClayPanel.Title>
+				}
+				displayType="secondary"
+				showCollapseIcon={true}
+			>
+				<ClayPanel.Body>
+					<p>
+						{+ticket.description
+							? ticket.description
+							: 'No description available.'}
+					</p>
+					{!ticket.assignee ? (
+						<ClayButton
+							displayType="secondary"
+							onClick={() => mutation.mutate(ticket)}
+							size="xs"
+						>
+							{isLoading ? (
+								<span className="inline-item inline-item-before">
+									<ClayLoadingIndicator
+										displayType="secondary"
+										size="sm"
+									/>
+								</span>
+							) : (
+								''
+							)}
+							Assign to Me
+						</ClayButton>
+					) : (
+						''
+					)}
+				</ClayPanel.Body>
+			</ClayPanel>
+		</div>
+	);
+};
+
+function assignToMe(ticket: Ticket) {
```

Comment:
It's better to be consistent, so I'd recommend just using arrow functions everywhere.